vault/resource_ad_secret_library.go

```diff
@@ -0,0 +1,232 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+)
+
+var (
+	adSecretBackendFromLibraryPathRegex = regexp.MustCompile("^(.+)/library/.+$")
```

Comment:
I don't think so, but is it possible to defeat this by having a set name that has `library/` in it? If that is possible then you should be able to fix by making the `(.+)` group into a non-greedy wildcard `(.+?)`.

Reply:
I don't think this is possible. Library names can't be nested.